### PR TITLE
Use time.Duration for time intervals

### PR DESCRIPTION
**Pull request by g41797**
PR for #83 
Fork was updated before changes

### CODE REVIEW

File: roc/receiver_test.go [hunk 1]
```diff
@@ -35,6 +35,46 @@ func TestReceiver_Open(t *testing.T) {
...
 			wantErr: newNativeErr("roc_receiver_open()", -1),
 		},
+		{
+			name: "invalid config.TargetLatency",
+			contextFunc: func() *Context {
+				ctx, err := OpenContext(makeContextConfig())
+				require.NoError(t, err)
+				return ctx
+			},
+			config:  makeReceiverConfig(WithTargetLatency(-1)),
```
Comment (gavv) [inline roc/receiver_test.go:45]: Regarding this bit:

```go
makeReceiverConfig(WithTargetLatency(-1))
```

I don't like a few things about this approach:

* test table become less self-contained; we need to look into implementation of functions like WithTargetLatency to see what they're doing

* we need to craft dozens of small functions, one per each field

* we complicate config construction by introducing concept of "options"

It seems there is a way to make things more simple and compact by modifying this approach a bit:

```go
name: "invalid config.TargetLatency",
contextFunc: func() *Context {
	ctx, err := OpenContext(makeContextConfig())
	require.NoError(t, err)
	return ctx
},
configFunc(): func() ReceiverConfig {
	cfg := makeReceiverConfig()
	cfg.TargetLatency = -1
	return cfg
},
```

File: roc/receiver_test.go [hunk 2]
```diff
@@ -35,6 +35,46 @@ func TestReceiver_Open(t *testing.T) {
...
+			wantErr: makeTDError(-1, "config.TargetLatency"),
```
Comment (gavv) [inline roc/receiver_test.go:46]: I think a small portion of copy-paste won't harm in this case and we can avoid introducing extra helper (makeTDError) to keep test table more descriptive:

```
wantErr: fmt.Errorf("invalid config.TargetLatency: %w",
		fmt.Errorf("unexpected negative duration: -1")
```

Like we do in all other tests.

**Review by gavv — COMMENTED** [review]
Thanks for update! A few more comments.

File: roc/convert.go [hunk 3]
```diff
@@ -37,3 +41,10 @@ func go2cBool(b bool) C.uint {
 	}
 	return 0
 }
+
+func go2cUnsignedDuration(d time.Duration) (ulonglong, error) {
```
Comment (gavv) [inline roc/convert.go:45]: Please add unit test for this func to `conver_test.go`

File: roc/receiver_test.go (hunk 2 above)
Comment (g41797) [reply roc/receiver_test.go:46]: no problem
after fix i'll update you via re-request review

File: roc/receiver_test.go (hunk 1 above)
Comment (g41797) [reply roc/receiver_test.go:45]: got the point

**Comment by g41797** [comment]
problems with environment
fix will take sometime

**Comment by gavv** [comment]
> problems with environment - ./receiver.go:313:13: could not determine kind of name for C.roc_receiver_set_reuseaddr
> 
> dejavu - in the past I solved this problem but don't remember how :-(
> 
> do you have some advise?
> 
> I will use re-request review to raise your attention

What version of libroc do you have installed on system? You can find it in `/usr/include/roc/version.h` or `/usr/local/include/roc/version.h`.

You can try installing latest version from `master` branch.

**Comment by g41797** [comment]
/usr/include/roc/version.h - 0.2.1 (github - 0.2.4 ) receiver.h doesn't include _roc_receiver_set_reuseaddr_

/usr/local/include - empty 

as far as I remember was installed from https://roc-streaming.org

after installation using zip from github version.h - 0.2.4

compilation - OK

Thanks a lot

**Pull request closed without merging**